espnet2/asr/encoder/contextual_block_conformer_encoder.py

```diff
@@ -314,8 +314,8 @@ def forward_train(
         mask_online = xs_pad.new_zeros(
             xs_pad.size(0), block_num, self.block_size + 2, self.block_size + 2
         )
-        mask_online.narrow(2, 1, self.block_size + 1).narrow(
-            3, 0, self.block_size + 1
+        mask_online.narrow(2, 0, self.block_size + 2).narrow(

```

Comment:
Did you need this modification to get the right result?  I believe this mask is for attention computation and it filter out unnecessary computation.  Each block consists of (previous context (1), frames (`self.block_size`), next context (1)) and input uses the first two and the output uses the last two, so it masks 0 to `self.block_size+1` for input and 1 to `self.block_size+2` for output.